hibernate-core/src/main/java/org/hibernate/internal/StatelessSessionImpl.java

```diff
@@ -298,28 +299,61 @@ public Object internalLoad(
 			// caller did not request forceful eager loading, see if we can create
 			// some form of proxy
 
+			final boolean allowBytecodeProxy = getFactory()
+					.getSessionFactoryOptions()
+					.isEnhancementAsProxyEnabled();
+
+			final boolean entityHasHibernateProxyFactory = persister.getEntityMetamodel()
+					.getTuplizer()
+					.getProxyFactory() != null;
+
```

Comment:
it can be moved inside the `if` statement

Reply:
OK, I'll do that.